src/integrations/watchers/indexable-post-type-archive-watcher.php

```diff
@@ -67,10 +67,6 @@ public function register_hooks() {
 	public function check_option( $old_value, $new_value ) {
 		$relevant_keys = [ 'title-ptarchive-', 'metadesc-ptarchive-', 'bctitle-ptarchive-', 'noindex-ptarchive-' ];
 
-		if ( ! is_array( $old_value ) || ! is_array( $new_value ) ) {
-			return;
-		}
```

Comment:
As discussed, we do want safety check around WP functionality. This is called straight from WP. And as you mentioned the `$old_value` might be false when this is the first save. We should be able to function properly if that is the case and there should be tests to cover that scenario.